python/lsst/meas/base/baseMeasurement.py

```diff
@@ -158,6 +167,10 @@ class BaseMeasurementConfig(lsst.pex.config.Config):
     where ``PluginBaseClass`` is the appropriate base class of the plugin
     (e.g., `SingleFramePlugin` or `ForcedPlugin`).
     """
+    def __new__(cls, *args, ignoreSlotPluginChecks=False, **kwargs):
+        instance = super().__new__(cls, *args, **kwargs)
```

Comment:
If this is deprecated we should raise a FutureWarning if this is set to True.

Reply:
If you are worried about tests spewing out deprecation warnings, I wouldn't be opposed to wrapping them in a warnings ignore.

Reply:
or using `with self.assertWarns()`

Reply:
I'll put in the warning, but if I recall correctly from Jim, there was no obvious path forward from here; the deprecation warning was mostly a "don't start using this" warning.

Reply:
Well, it should be warning of some sort, and we can catch that in the tests.  That way anybody else who tries this funny business can't say they weren't warned.

Reply:
Looks like we have to ignore/catch them in the tests: when I ran scons, the tests all raised!

Reply:
`assertWarns` is going to complain if the warning does not appear so are there cases that do issue warnings and cases that don't?

Reply:
What do you mean?

Reply:
Are you asking me or John? I interpreted John's "the tests all raised" as meaning that some tests did issue warnings that the assertWarns caught but other tested wrapped in assertWarns did not issue warnings and so triggered test failures. If you aren't really sure which bit of code is going to warn or not then globally ignoring warnings for the test is the only choice (or else use assertWarns and then issue a warning of your own for it to catch).

Reply:
I was asking John, but I think you have the correct interpretation.  But I don't know why some tests wouldn't be warning.  I'll leave this ti John.

Reply:
My point is that this `FutureWarning` raised and failed the tests. I wasn't aware that we'd switched this package to fail on warnings.

Reply:
As far as I'm aware no package is configured to fail tests if a warning is issued (all our packages would fail). Can you please show the output you get when your test fails (since `assertWarns` isn't even in play).

Reply:
`main` of `meas_base` issues 3 FutureWarning message already:

```
tests/test_MeasureSources.py::MeasureSourcesTestCase::testPeakLikelihoodFlux
  /Users/timj/work/lsstsw/build/meas_base/tests/test_MeasureSources.py:113: FutureWarning: Default position argument overload is deprecated and will be removed in version 24.0.  Please explicitly specify a position.
    psfKernel = psf.getLocalKernel()

tests/test_MeasureSources.py::MeasureSourcesTestCase::testPeakLikelihoodFlux
  /Users/timj/work/lsstsw/build/meas_base/tests/test_MeasureSources.py:114: FutureWarning: Default position argument overload is deprecated and will be removed in version 24.0.  Please explicitly specify a position.
    psfImage = psf.computeKernelImage()

tests/test_diaCalculation.py::TestDiaCalcluation::testRunUnindexed
  /Users/timj/work/lsstsw/build/meas_base/tests/test_diaCalculation.py:213: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
    unindexedDiaSources = unindexedDiaSources.append(
```

Reply:
```____________________ PsfFluxTestCase.testSingleFramePlugin _____________________
[gw4] linux -- Python 3.10.5 /home/parejkoj/lsst/lsstsw/miniconda/envs/lsst-scipipe-4.1.0/bin/python3.10

self = <test_PsfFlux.PsfFluxTestCase testMethod=testSingleFramePlugin>

    def testSingleFramePlugin(self):
>       task = self.makeSingleFrameMeasurementTask("base_PsfFlux")

tests/test_PsfFlux.py:215:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python/lsst/meas/base/tests.py:666: in makeSingleFrameMeasurementTask
    config = self.makeSingleFrameMeasurementConfig(plugin=plugin, dependencies=dependencies)
python/lsst/meas/base/tests.py:624: in makeSingleFrameMeasurementConfig
    config = SingleFrameMeasurementTask.ConfigClass(ignoreSlotPluginChecks=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'lsst.meas.base.sfm.SingleFrameMeasurementConfig'>
ignoreSlotPluginChecks = True, args = (), kwargs = {}
instance = lsst.meas.base.sfm.SingleFrameMeasurementConfig(slots={'centroid': 'base_SdssCentroid', 'shape': 'base_SdssShape', 'ps...am': 0.0}, 'test_NoiseReplacer': {'doMeasure': True}, 'test_LoggingPlugin': {'doMeasure': True}}}, loggingInterval=600)
msg = 'ignoreSlotPluginChecks is deprecated and should only be used in tests. No removal date has been set yet.'

    def __new__(cls, *args, ignoreSlotPluginChecks=False, **kwargs):
        instance = super().__new__(cls, *args, **kwargs)
        if ignoreSlotPluginChecks:
            msg = ("ignoreSlotPluginChecks is deprecated and should only be used in tests."
                   " No removal date has been set yet.")
>           raise FutureWarning(msg)
E           FutureWarning: ignoreSlotPluginChecks is deprecated and should only be used in tests. No removal date has been set yet.

python/lsst/meas/base/baseMeasurement.py:175: FutureWarning```

Reply:
It's supposed to be a `warnings.warn("Message", category=FutureWarning, stacklevel=2)` not a `raise`.

Reply:
Oh, I need to do `warnings.warn(msg, FutureWarning)`. Sorry about the chatter.

Reply:
Yes, `stacklevel=2` is the polite option to add as well because it means that the caller gets told the line of theirs that caused the warning to trigger, rather than the line in meas_base.

Reply:
If that's a thing we should do in warnings, please add it to the dev guide deprecations section, possibly [here in package deprecation](https://developer.lsst.io/stack/deprecating-interfaces.html#package-deprecation), but maybe we should have a paragraph in the python section above that about deprecating function call signatures too?